chutney/ui/src/app/core/services/sso.service.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-FileCopyrightText: 2017-2024 Enedis
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { AuthConfig, OAuthService } from 'angular-oauth2-oidc';
+import { environment } from '@env/environment';
+import {Observable, map, tap} from 'rxjs';
+import { Injectable } from '@angular/core';
+
+interface SsoAuthConfig {
+  issuer: string,
+  clientId: string,
+  clientSecret: string,
+  responseType: string,
+  scope: string,
+  redirectBaseUrl: string,
+  ssoProviderName: string,
+  oidc: boolean
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SsoService {
+
+  private resourceUrl = '/api/v1/sso/config';
+
+  private ssoConfig: SsoAuthConfig
+
+
+  constructor(private oauthService: OAuthService, private http: HttpClient) {}
+
+  fetchSsoConfig(): void {
+    this.http.get<SsoAuthConfig>(environment.backend + this.resourceUrl).pipe(
+        map(ssoConfig => {
+          this.ssoConfig = ssoConfig
+          return {
+            issuer: ssoConfig.issuer,
+            clientId: ssoConfig.clientId,
+            responseType: ssoConfig.responseType,
+            scope: ssoConfig.scope,
+            redirectUri: ssoConfig.redirectBaseUrl + '/',
+            dummyClientSecret: ssoConfig.clientSecret,
+            oidc: ssoConfig.oidc
+          }
+        }),
+        tap(ssoConfig => {
+          this.oauthService.configure(ssoConfig)
+          this.oauthService.loadDiscoveryDocumentAndTryLogin();
```

Comment:
Is this mandatory ? Could we catch error if provider is not here ?

Reply:
Yes, this is mandatory